File: killrvideo-service-videocatalog/src/main/java/com/killrvideo/service/video/dao/VideoCatalogDaoConfig.java
package com.killrvideo.service.video.dao;

import com.datastax.oss.driver.api.core.CqlSession;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class VideoCatalogDaoConfig {
    @Bean
    public VideoCatalogMapper videoCatalogMapper(CqlSession session) {
        return VideoCatalogMapper.build(session).build();
    }
}

File: killrvideo-service-videocatalog/src/main/kotlin/com/killrvideo/service/video/dao/VideoCatalogDaoConfig.kt
package com.killrvideo.service.video.dao

import com.datastax.oss.driver.api.core.CqlSession
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class VideoCatalogDaoConfig {
    @Bean
    open fun videoCatalogMapper(session: CqlSession): VideoCatalogMapper {
        return VideoCatalogMapperBuilder(session).build()
    }
}
